src/Data/Bin/Bin0.agda

```diff
@@ -0,0 +1,655 @@
+  fromℕ (toℕ (a + b))        ≡⟨ cong fromℕ (toℕ+homo a b) ⟩
+  fromℕ (toℕ a +' toℕ b)    ≡⟨ cong fromℕ (+'-comm (toℕ a) (toℕ b)) ⟩
+  fromℕ (toℕ b +' toℕ a)    ≡⟨ cong fromℕ (sym (toℕ+homo b a)) ⟩
+  fromℕ (toℕ (b + a))        ≡⟨ fromℕ∘toℕ (b + a) ⟩
+  b + a                        ∎
+
++-assoc :  Associative _+_
++-assoc a b c = begin
+  (a + b) + c                    ≡⟨ sym (fromℕ∘toℕ ((a + b) + c)) ⟩
+  fromℕ (toℕ ((a + b) + c))    ≡⟨ cong fromℕ (toℕ+homo (a + b) c) ⟩
+  fromℕ (toℕ (a + b) +' cN)    ≡⟨ cong (fromℕ ∘ (_+' cN)) (toℕ+homo a b) ⟩
+  fromℕ ((aN +' bN) +' cN)      ≡⟨ cong fromℕ (+'-assoc aN bN cN) ⟩
+  fromℕ (aN +' (bN +' cN))      ≡⟨ cong (fromℕ ∘ (aN +'_))
+                                                  (sym (toℕ+homo b c)) ⟩
+  fromℕ (aN +' toℕ (b + c))    ≡⟨ cong fromℕ (sym (toℕ+homo a (b + c))) ⟩
+  fromℕ (toℕ (a + (b + c)))    ≡⟨ fromℕ∘toℕ (a + (b + c)) ⟩
+  (a + (b + c))                  ∎
+  where
+  aN = toℕ a;   bN = toℕ b;   cN = toℕ c
+
+
++-lCancel :  ∀ x y z → x + y ≡ x + z → y ≡ z      -- the proof is via toℕ
++-lCancel x y z x+y≡x+z = begin
+  y          ≡⟨ sym (fromℕ∘toℕ y) ⟩
+  fromℕ m   ≡⟨ cong fromℕ m≡n ⟩
+  fromℕ n   ≡⟨ fromℕ∘toℕ z ⟩
+  z          ∎
+  where
+  k = toℕ x;  m = toℕ y;  n = toℕ z
+
+  eq = begin
+    k +' m          ≡⟨ sym (toℕ+homo x y) ⟩
+    toℕ (x + y)    ≡⟨ cong toℕ x+y≡x+z ⟩
+    toℕ (x + z)    ≡⟨ toℕ+homo x z ⟩
+    k +' n          ∎
+
+  m≡n = begin
+    m                ≡⟨ sym (NatP.m+n∸n≡m m k)⟩
+    (m +' k) ∸' k    ≡⟨ cong (_∸' k) (+'-comm m k) ⟩
+    (k +' m) ∸' k    ≡⟨ cong (_∸' k) eq ⟩
+    (k +' n) ∸' k    ≡⟨ cong (_∸' k) (+'-comm k n) ⟩
+    (n +' k) ∸' k    ≡⟨ NatP.m+n∸n≡m n k ⟩
+    n                ∎
+
++-rCancel :  ∀ x y z → y + x ≡ z + x → y ≡ z
++-rCancel x y z y+x≡z+x =  +-lCancel x y z x+y≡x+z
+  where
+  x+y≡x+z = begin
+    x + y   ≡⟨ +-comm x y ⟩
+    y + x   ≡⟨ y+x≡z+x ⟩
+    z + x   ≡⟨ +-comm z x ⟩
+    x + z   ∎
+
+decEquivalence :  IsDecEquivalence (_≡_ {A = Bin})
+decEquivalence =  record{ isEquivalence = isEquivalence;  _≟_ = _≟_ }
+
+decSetoid : DecSetoid 0ℓ 0ℓ
+decSetoid = record{ Carrier          =  Bin
+                  ; _≈_              =  _≡_
+                  ; isDecEquivalence =  decEquivalence }
+
++-isMagma :  IsMagma _≡_ _+_
++-isMagma =  record{ isEquivalence =  isEquivalence
+                   ; ∙-cong        =  cong₂ _+_
+                   }
+
++-magma :  Magma 0ℓ 0ℓ
++-magma =  record{ Carrier = Bin
+                 ; _≈_     = _≡_
+                 ; _∙_     = _+_
+                 ; isMagma = +-isMagma
+                 }
+
++-isSemigroup : IsSemigroup _≡_ _+_
++-isSemigroup = record{ isMagma = +-isMagma;  assoc = +-assoc }
+
++-semigroup : Semigroup 0ℓ 0ℓ
++-semigroup = record{ Carrier     = Bin
+                    ; _≈_         = _≡_
+                    ; _∙_         = _+_
+                    ; isSemigroup = +-isSemigroup }
+
++-isMonoid : IsMonoid _≡_ _+_ 0#
++-isMonoid = record{ isSemigroup = +-isSemigroup;
```

Comment:
The name of the `IsMonoid` proof should mention the neutral element. Cf. [`+-0-isMonoid : IsMonoid _+_ 0`](https://agda.github.io/agda-stdlib/Data.Nat.Properties.html#12697)

Reply:
The same goes for the other proofs below.